gallery_dl/extractor/deviantart.py

```diff
@@ -1475,11 +1490,26 @@ def _pagination(self, endpoint, params,
                             "oauth:deviantart' and follow the instructions to "
                             "be able to access them.")
                 # "statusid" cannot be used instead
-                if results and "deviationid" in results[0]:
-                    if self.metadata:
-                        self._metadata(results)
-                    if self.folders:
-                        self._folders(results)
+                if results:
+                    if "deviationid" in results[0]:
+                        if self.metadata:
+                            self._metadata(results)
+                        if self.folders:
+                            self._folders(results)
+                    else:
+                        # attempt to fix "deleted" deviations
+                        for result in results:
+                            for item in result.get("items") or ():
+                                if "deviation" not in item or \
+                                        not item["deviation"]["is_deleted"]:
+                                    continue
+                                patch = self._call(
+                                    "/deviation/" +
+                                    item["deviation"]["deviationid"],
+                                    fatal=False)
+                                if patch:
+                                    item["deviation"] = patch
+
```

Comment:
Just a thought, but maybe it'd be better to use a separate `_pagination` method for status items? Or at least put fixing deleted deviations in its own method.

I admit to having done this myself (plenty of times?) , but it hurts even me seeing ~50 spaces of indentation.

Reply:
The implementation here will be different if combined with [this branch](https://github.com/ClosedPort22/gallery-dl/blob/da-status-expand/gallery_dl/extractor/deviantart.py), which is almost done (I think). I'll try to refactor this section to make it look nicer though.